pandas/tests/groupby/test_groupby.py

```diff
@@ -23,6 +23,38 @@
 import pandas.util.testing as tm
 import pandas as pd
 
+from pandas._libs import reduction
+
+
+def demean(arr):
+    return arr - arr.mean()
+
+
+def test_apply_frame_axis0_runs_each_group_once(skip_first_index=True):
+    class CountCalls:
+        def __init__(self):
+            self.counter = 0
+
+        def f(self, x):
+            self.counter += 1
+            return demean(x)
```

Comment:
What about

```
from itertools import count
f = count()
...
assert f() = 3
```
?

Reply:
Good idea. I'm not familiar with using count in this way without `__next__` method. Could you explain a bit more?

I tried the following, and it works:

```python
    from itertools import count
    cc = count()
    f = lambda x: cc.__next__()  # need a function with at least one input, otherwise there will be an error
    ...
    assert f(None) == 2
```

Reply:
Sorry, my mistake again.

I think your solution only works with python 3.*, while
```
f = lambda x : next(cc)
```
should also work in 2.7. And yes, it will be ``==2``, not ``==3``. Maybe ``assert next(cc) == 2`` is slightly cleaner though.

Reply:
In the past, pytest's assertion rewriting had trouble with statements that have side-effects like this. I'd recommend

```python
result = next(cc)
assert result == 2
```